src/main/java/ua/softserveinc/tc/service/impl/BookingServiceImpl.java

```diff
@@ -164,18 +164,40 @@ public Boolean checkForDuplicateBookingSingle(BookingDto bookingDto) {
         return isDuplicate;
     }
 
-
+    /**
+     * The method finds out is there available space in
+     * the rooms for given listDTO.
+     * @param listDTO list of BookingDto
+     * @return true if there is available places in the room
+     */
+    private boolean isAvailabilePlacesInTheRoom(List<BookingDto> listDTO) {
+        int availablePlaces = 0;
+        int needPlaces = 1;
+        Date theSameDay = null;
+        for(BookingDto bdto : listDTO) {
+            if(bdto.getDateStartTime().equals(theSameDay)) {
+                needPlaces++;
+                continue;
+            } else if (theSameDay != null && availablePlaces < needPlaces)
+                return false;
+            needPlaces = 1;
+
+            theSameDay = bdto.getDateStartTime();
+            availablePlaces = roomService.getAvailableSpaceForPeriod(
+                    bdto.getDateStartTime(),
+                    bdto.getDateEndTime(),
+                    bdto.getRoom());
+        }
+        return availablePlaces >= needPlaces;
+    }
 
     @Override
     public List<BookingDto> persistBookingsFromDtoAndSetId(List<BookingDto> listDTO) {
-        BookingDto bdto = listDTO.get(0);
-
-        int available = roomService.getAvailableSpaceForPeriod(
-                bdto.getDateStartTime(),
-                bdto.getDateEndTime(),
-                bdto.getRoom());
+        if(listDTO == null)
+            throw new RuntimeException("listDTO is null");
```

Comment:
This is not good practice, because we hide our problem and it going to be harder to find it later. RuntimeException is not checked, so it is not must be caught. It is very bad style to hide problem in empty list. Program will be fail in any case.